autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/FastICA.py

```diff
@@ -0,0 +1,102 @@
+            fun=self.fun, whiten=self.whiten, random_state=self.random_state)
+
+        return self
+
+    @staticmethod
+    def get_hyperparameter_search_space(
+        dataset_properties: Optional[Dict[str, BaseDatasetPropertiesType]] = None,
+        n_components: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='n_components',
+                                                                            value_range=(0.5, 0.9),
+                                                                            default_value=0.5,
+                                                                            ),
+        algorithm: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='algorithm',
+                                                                         value_range=('parallel', 'deflation'),
+                                                                         default_value='parallel',
+                                                                         ),
+        whiten: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='whiten',
+                                                                      value_range=(True, False),
+                                                                      default_value=False,
+                                                                      ),
+        fun: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='fun',
+                                                                   value_range=('logcosh', 'exp', 'cube'),
+                                                                   default_value='logcosh',
+                                                                   ),
+    ) -> ConfigurationSpace:
+        if dataset_properties is not None:
+            n_features = len(dataset_properties['numerical_columns']) if isinstance(
+                dataset_properties['numerical_columns'], List) else 0
+            if n_features == 1:
+                log = False
+            else:
+                log = n_components.log
+            n_components = HyperparameterSearchSpace(hyperparameter='n_components',
+                                                     value_range=(
+                                                         floor(float(n_components.value_range[0]) * n_features),
+                                                         ceil(float(n_components.value_range[1]) * n_features)),
+                                                     default_value=ceil(float(n_components.default_value) * n_features),
+                                                     log=log)
+        else:
+            n_components = HyperparameterSearchSpace(hyperparameter='n_components',
+                                                     value_range=(10, 2000),
+                                                     default_value=100,
+                                                     log=n_components.log)
```

Comment:
Can we make a method to calculate `value_range`, `default_value`, and `log`?

Reply:
done